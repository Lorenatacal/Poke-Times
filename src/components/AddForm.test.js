import React, { Component } from 'react';
import AddFormConnected, { AddForm } from './AddForm';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

Enzyme.configure({ adapter: new Adapter() });

test("AddForm should render correctly", () => {
    const wrapper = Enzyme.shallow(<AddForm />);
    expect(toJson(wrapper)).toMatchSnapshot();
})

test("AddForm should call addPost when the user inputs title, body and clicks submit", () => {
    const spy = jest.fn();
    // pass the props as an object that you spread in the component
    const wrapper = Enzyme.shallow(<AddForm addPostCallback={spy} />);
    const userTitleInput = wrapper.find('[data-name="userTitle"]');
    userTitleInput.simulate('change', { target: { title: "New Post" } });
    const submitButton = wrapper.find('[data-name="submitForm"]');
    submitButton.simulate('click', { preventDefault() {} });

    expect(spy).toHaveBeenCalled();
})