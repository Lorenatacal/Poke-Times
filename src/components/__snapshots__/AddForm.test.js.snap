// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddForm should render correctly 1`] = `
<div
  className="center post card"
>
  <form
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      placeholder="Your title"
      value=""
    />
    <textarea
      onChange={[Function]}
      placeholder="Insert the post body"
      value=""
    />
    <button
      onClick={[Function]}
      type="Submit"
    >
      Submit 
    </button>
  </form>
</div>
`;
